docs/source/API/batched/dense/KokkosBatched_ApplyHouseholder.rst

```diff
@@ -0,0 +1,326 @@
+KokkosBatched::ApplyHouseholder
+###############################
+
+Defined in header: :code:`KokkosBatched_ApplyHouseholder_Decl.hpp`
+
+.. code-block:: c++
+
+    // Serial version
+    template <typename ArgSide>
+    struct SerialApplyHouseholder {
+      template <typename uViewType, typename tauViewType, typename AViewType, typename wViewType>
+      KOKKOS_INLINE_FUNCTION
+      static int
+      invoke(const uViewType& u2,
+             const tauViewType& tau,
+             const AViewType& A,
+             const wViewType& w);
+    };
+    
+    // Team Vector version
+    template <typename MemberType, typename ArgSide>
+    struct TeamVectorApplyHouseholder {
+      template <typename uViewType, typename tauViewType, typename AViewType, typename wViewType>
+      KOKKOS_INLINE_FUNCTION
+      static int
+      invoke(const MemberType& member,
+             const uViewType& u2,
+             const tauViewType& tau,
+             const AViewType& A,
+             const wViewType& w);
+    };
+
+The ``ApplyHouseholder`` operation applies a Householder transformation to a matrix. This is a fundamental building block for many linear algebra operations, including QR factorization.
+
+A Householder transformation is defined by a Householder vector ``u`` and a scalar ``tau``. It applies the transformation matrix :math:`H = I - \tau u u^T` (for real matrices) or :math:`H = I - \tau u u^H` (for complex matrices) to a target matrix.
+
+Mathematically, when applied from the left side:
+
+.. math::
+
+    A := H \cdot A = (I - \tau u u^T) \cdot A
+
+When applied from the right side:
+
+.. math::
+
+    A := A \cdot H = A \cdot (I - \tau u u^T)
+
+where:
+
+- :math:`I` is the identity matrix
+- :math:`u` is the Householder vector
```

Comment:
I am not sure a Householder vector is really a thing, I would argue that it is more interesting to explain what it is with a formula such as `u_x=x-||x||e_0`